internal/pkg/link/link.go

```diff
@@ -0,0 +1,151 @@
+// -----------------------------------------------------------------------------
```

Comment:
We could probably stand to add some more logging to this file to give insight on what is happening.